### PR TITLE
Reconnection fixes

### PR DESCRIPTION
**Pull request by AeroNotix**

### CODE REVIEW

File: src/folsomite_graphite_client.erl [hunk 1]
```diff
@@ -60,11 +60,12 @@ handle_cast(Cast, State) ->
     unexpected(cast, Cast),
     {noreply, State}.
 
-handle_info(connect, #state{host=Host, port=Port, reconnect_time=RCT} = State) ->
+handle_info(connect, #state{host=Host, port=Port, reconnect_time=RCT, socket=OldSock} = State) ->
     error_logger:info_msg("Folsomite attempting to reconnect: ~s:~p", [Host, Port]),
     Opts = [binary, {active, false}, {send_timeout, ?SEND_TIMEOUT}],
     case gen_tcp:connect(Host, Port, Opts, ?CONNECT_TIMEOUT) of
         {ok, Sock} ->
+            ok = gen_tcp:close(OldSock),
```
Comment (AeroNotix) [inline src/folsomite_graphite_client.erl]: :footprints:

**Comment by AeroNotix** [comment]
handled an undefined tcp sock when closing

**Comment by puzza007** [comment]
Ta! Want me to push a new version to hex.pm?

**Comment by AeroNotix** [comment]
what about cloudi folsomite? Do we have enough cycles to run a full week's worth of releases?

**Comment by puzza007** [comment]
We'd have to carefully consider the downstream impact of such a change on the CloudI ecosystem.

**Pull request merged**